src/client/Client.cc

```diff
@@ -2834,8 +2846,8 @@ void Client::send_reconnect(MetaSession *session)
     }
   }
 
-  early_kick_flushing_caps(session);
-
+  if (!multi)
```

Comment:
````
bool can_multi_reconnect = session->mds_features.test(CEPHFS_FEATURE_MULTI_RECONNECT);
...
if (!can_multi_reconnect)
````

i.e., do not limit the version just because we're not sending multiple reconnect messages.